docs-site/content/0.19.0/guide/mongodb-full-text-search.md

```diff
@@ -0,0 +1,158 @@
+```bash
+mongod --port "PORT" --dbpath "YOUR_DB_DATA_PATH" --replSet "REPLICA_SET_INSTANCE_NAME"
+```
+
+Check the status of replica set issuing the command `rs.status()` in mongo shell.
+
+## Step 3: Open a Change Stream
+
+We can open a change stream for MongoDB Replica Set from any of the data-bearing members.
+
+Example,
+```js
+const uri = '<MongoDB-URI>';
+const mongodbOptions = {
+    useNewUrlParser: true,
+    useUnifiedTopology: true
+}
+const client = new MongoClient(uri, mongodbOptions);
+await client.connect();
+const collection = client.db("sample").collection("books");
+const changeStream = collection.watch();
+changeStream.on('change', (next) => {
+    // process next document
+});
+```
+
+## Step 4: Create a Typesense Collection
+
+To use Typesense, we first need to create a client. Typesense supports multiple API clients including Javascript, Python, Ruby, PHP etc.
+
+To create the Javascript client, you need the API key of the Typesense server:
+
+```js
+import Typesense from 'typesense'
+
+let typesense = new Typesense.Client({
+  'nodes': [{
+    'host': 'localhost',
+    'port': '8108',
+    'protocol': 'http'
+  }],
+  'apiKey': '<API_KEY>',
+  'connectionTimeoutSeconds': 2
+})
+```
+Next, we will create a collection. A collection needs a schema, that represents how a douments would look like.
+
+```js
+let schema = {
+    'name': 'books',
+    'fields': [
+        { 'name': 'id', 'type': 'string', 'facet': false },
+        { 'name': 'name', 'type': 'string','facet': false },
+        { 'name': 'author', 'type': 'string', 'facet': false },
+        { 'name': 'year', 'type': 'int32', 'facet': true },
+    ],
+    'default_sorting_field': 'year',
+}
+await typesense.collections().create(schema);
+```
+
+## Step 5: Index documents to Typesense
+
+Next, we'll write functions to listen to change streams from MongoDB and write the changes to Typesense.
+
+A example MongoDB change streams response,
+
+```js
+{
+  _id: {
+    _data: '826062978E000000012B022C0100296E5'
+  },
+  operationType: 'insert',
+  clusterTime: Timestamp { _bsontype: 'Timestamp', low_: 1, high_: 1617074062 },
+  fullDocument: {
+    _id: 6062978e06e4444ef0c7f16a,
+    name: 'Davinci Code',
+    author: 'Dan Brown',
+    year: 2003
+  },
+  ns: { db: 'sample', coll: 'books' },
+  documentKey: { _id: 6062978e06e4444ef0c7f16a }
+}
+{
+  _id: {
+    _data: '826062978E000000032B022C0100296E5'
+  },
+  operationType: 'update',
+  clusterTime: Timestamp { _bsontype: 'Timestamp', low_: 3, high_: 1617074062 },
+  ns: { db: 'sample', coll: 'books' },
+  documentKey: { _id: 6062978e06e4444ef0c7f16a },
+  updateDescription: { updatedFields: { year: 2000 }, removedFields: [] }
+}
+{
+  _id: {
+    _data: '826062978E000000072B022C0100296E5'
+  },
+  operationType: 'delete',
+  clusterTime: Timestamp { _bsontype: 'Timestamp', low_: 7, high_: 1617074062 },
+  ns: { db: 'sample', coll: 'books' },
+  documentKey: { _id: 6062978e06e4444ef0c7f16c }
+}
+```
+A example function,
```

Comment:
Could you expand on this? An "example function" sounds vague. Could you explain in words what this snippet does?